lib/jellyfish_web/api_spec/component.ex

```diff
@@ -29,36 +28,26 @@ defmodule JellyfishWeb.ApiSpec.Component do
       description: "Component-specific options",
       type: :object,
       oneOf: [
-        RTSP
-      ],
-      nullable: true
-    })
-  end
-
-  defmodule Metadata do
-    @moduledoc false
-
-    require OpenApiSpex
-
-    OpenApiSpex.schema(%{
-      title: "ComponentMetadata",
-      description: "Component-specific metadata",
-      type: :object,
-      properties: %{
-        playable: %Schema{type: :boolean}
-      }
+        HLS.Options,
+        RTSP.Options
+      ]
     })
   end
 
   OpenApiSpex.schema(%{
     title: "Component",
     description: "Describes component",
     type: :object,
-    properties: %{
-      id: %Schema{type: :string, description: "Assigned component id", example: "component-1"},
-      type: Type,
-      metadata: Metadata
-    },
-    required: [:id, :type, :metadata]
+    oneOf: [
+      HLS,
+      RTSP
+    ],
+    discriminator: %OpenApiSpex.Discriminator{
+      propertyName: "type",
+      mapping: %{
+        "hls" => HLS,
+        "rtsp" => RTSP
+      }
+    }
```

Comment:
Without the discriminator, the `component_controller` tests fail at `assert_response_schema` due to "multiple schemas match". That's because there are no fields in RTSP component metadata, and an empty map matches every other map.